### PR TITLE
test: Rekor monitor log test

### PR DESCRIPTION
**Pull request by petrpinkas**
## Summary by Sourcery

Add end-to-end integration tests for the Rekor monitor log and introduce helper functions to retrieve and parse monitor metrics.

Enhancements:
- Implement TAS Rekor helper functions to fetch, parse, and return monitor metrics via the Pod proxy for use in tests.

Tests:
- Add integration test suite verifying monitor pod deployment, volume mounting, initial empty log behavior, metric increments, consistency verification after cosign signing, and subtle checkpoint corruption detection.

___

### **PR Type**
Tests

___

### **Description**
- Add comprehensive E2E tests for Rekor monitor log functionality

- Verify monitor pod deployment and persistent volume mounting

- Test empty log handling and consistency verification behavior

- Implement corruption detection tests with subtle hash modifications

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Test Setup"] --> B["Monitor Pod Deployment"]
  B --> C["Empty Log Verification"]
  C --> D["Metrics Validation"]
  D --> E["Cosign Signing Test"]
  E --> F["Corruption Detection"]
  F --> G["Hash Modification Test"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>rekor_monitor_log_test.go</strong><dd><code>Complete E2E test suite for Rekor monitor</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/e2e/rekor_monitor_log_test.go

<ul><li>Implements comprehensive E2E test suite for Rekor monitor <br>functionality<br> <li> Tests pod deployment, persistent volume mounting, and readiness checks<br> <li> Validates empty log handling without consistency verification<br> <li> Verifies metrics behavior with increasing totals and zero failures<br> <li> Tests corruption detection through subtle checkpoint file hash <br>modifications</ul>

</details>

  </td>
  <td><a href="https://github.com/securesign/secure-sign-operator/pull/1325/files#diff-f49647ebed959950ff6ad3ab8d80c2a29d33865960bedb584f87c36884242e7e">+259/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>monitor.go</strong><dd><code>Monitor metrics utility functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/e2e/support/tas/rekor/monitor.go

<ul><li>Adds utility functions for retrieving monitor metrics from pods<br> <li> Implements metric parsing for <code>log_index_verification_total</code> and <br><code>log_index_verification_failure</code><br> <li> Provides helper functions to locate and interact with monitor pods</ul>

</details>

  </td>
  <td><a href="https://github.com/securesign/secure-sign-operator/pull/1325/files#diff-041c606f32a3043934a34f0d25264bce587c0641c65017f528201c5115a36386">+78/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: test/e2e/rekor_monitor_log_test.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	cli, _ := support.CreateClient()
+
+	var (
+		namespace       *v1.Namespace
+		s               *v1alpha1.Securesign
+		signedImageName string
+		rekorMonitorPod v1.Pod
+	)
+
+	execOnMonitorPodWithOutput := func(command ...string) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (petrpinkas) [reply test/e2e/rekor_monitor_log_test.go:38]: Thx for hint. Done.
